src/main/kotlin/fi/metatavu/lipsanen/projects/milestones/tasks/TaskController.kt

```diff
@@ -138,12 +144,18 @@ class TaskController {
             milestone.endDate = newTask.endDate
         }
 
-        existingTask.startDate = newTask.startDate
-        existingTask.endDate = newTask.endDate
-
-        existingTask.status = newTask.status    // verification if updates are required is done in the api impl
-        existingTask.name = newTask.name
-        existingTask.lastModifierId = userId
+        with(existingTask) {
+            startDate = newTask.startDate
+            endDate = newTask.endDate
+            status = newTask.status
+            name = newTask.name
+            lastModifierId = userId
+            assigneeIds = newTask.assigneeIds ?: emptyList()
+            userRole = newTask.userRole ?: UserRole.USER
+            estimatedDuration = newTask.estimatedDuration ?: ""
+            estimatedReadiness = newTask.estimatedReadiness ?: ""
+            attachmentUrls = newTask.attachmentUrls ?: emptyList()
```

Comment:
If these are nullable fields what's the need to assign empty values? They can be null